src/mbgl/style/expression/compound_expression.cpp

```diff
@@ -246,12 +246,10 @@ optional<double> featurePropertyAsDouble(const EvaluationContext& params, const
     assert(params.feature);
     auto property = params.feature->getValue(key);
     if (!property) return {};
-    return property->match(
-        [](double value) { return value; },
-        [](uint64_t value) { return optional<double>(static_cast<double>(value)); },
-        [](int64_t value) { return optional<double>(static_cast<double>(value)); },
-        [](auto) { return optional<double>(); }
-    );
+    return property->match([](double value) { return value; },
+                           [](uint64_t value) -> optional<double> { return {static_cast<double>(value)}; },
+                           [](int64_t value) -> optional<double> { return {static_cast<double>(value)}; },
+                           [](auto) -> optional<double> { return {}; });
```

Comment:
const auto&

Reply:
there are other places where we make unnecessary copies :(
maybe replace all of them in this PR?
```
[](auto) -> [](const auto&)
[](std::string) -> [](const std::string&)
```

Reply:
Found two more places like this